sw/airborne/modules/sonar/sonar_bebop.c

```diff
@@ -205,3 +226,37 @@ void *sonar_bebop_read(void *data __attribute__((unused)))
   }
   return NULL;
 }
+
+float sonar_filter_gate(float distance_sonar)
+{
+  float distance;
+  current_distance = distance_sonar;
+  diff_pre_cur = current_distance - previous_distance;
+  if (diff_pre_cur < -0.4) {
+    z0 = previous_distance;
+    through_gate_green_light = 1;
+    counter_temp3 = 0;
+    time_temp3 = 0;
+  }
+  if (diff_pre_cur > 0.4 || time_temp3 > 1.0) {
+
+    through_gate_green_light = 0;
+  }
+
+  if (through_gate_green_light == 1) {
+    distance = z0;
+  } else {
+    distance = distance_sonar;
+  }
+  previous_distance = current_distance;
+  if (distance > 5) {
```

Comment:
This is very strange hard coded value. There are already defines in the ins to only consider sonar values in a certain range.